ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -914,12 +916,12 @@ protected void maybeProcessCallbacks() {
      */
     protected ByteBuf decompressContent(ByteBuf sourceBuffer) {
       if (!isChunkCompressed) {
-        // Blob is not compressed.  Return source buffer.
+        // Blob is not compressed.  Return source buffer and increase the counter.
```

Comment:
Shall we call sourceBuffer.release here since we add refer count before calling this function?

Reply:
we don't want to release source buffer since we are going to use source buffer later now that the chunk is not compressed.